### ISSUE TITLE
Implement Separating Axis Theorem

### ISSUE THREAD

**Issue by trommlbomml** [MonoGame/MonoGame#4378]
As I think it is relatively easy to solve for AABB and Sphere, this would really help to be in the main framework.

**Comment by tomspilman** [comment]
What new API are you specifically proposing?

We already have `BoundingBox` vs `BoundingSphere` test:

  https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/BoundingBox.cs#L110

We also have `BoundingSphere` vs `BoundingBox`:

https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/BoundingSphere.cs#L74

Both tell you if the primitive is fully contained, overlapping, or not touching at all.

**Comment by WardBenjamin** [comment]
If it's easy, why not do it yourself :stuck_out_tongue: 

I'm sure that the Monogame maintainers would be much more inclined to do this if they were considering a pull request vs extra room for them. The github issues are not really a feature request.

**Comment by tomspilman** [comment]
> The github issues are not really a feature request.

Well not entirely true...  it is fine to make a feature request here.

However as @WardBenjamin says it is a million times better to have someone submit a PR with the feature.

More importaintly for this issue is understanding specifically what @trommlbomml is looking for.

**Comment by nivan22** [comment]
As I think itimerne...?

**Comment by trommlbomml** [comment]
I just created the issue of course I will provide a solution. I am interesting especially in the closest contact point for proper movements. This is not possible currently but very handy.

And yes, when my implementation is finished I will provide the solution for monogame for sure. But I started an issue to maybe there are reasons to not include this solution.

**Comment by tomspilman** [comment]
>  I am interesting especially in the closest contact point for proper movements. 

So what you want is a closest point API in `BoundingBox` and `BoundingSphere`?

**Comment by trommlbomml** [comment]
Well I think more than a "what I have to track back so the boundings are not colliding any more". I found a term called MVT (Minimum Translation Vector), so the collision shapes get out of collision from each other.
The cool thing is, this value is part of finding the separating axis theorem and is fast in executing, espeically for axis aligned BoundingBoxes.

Furthermore, this axis can be projected to the moving vector of the objects, as the separated axis theorem just returns for the tested axis independent from the movement direction.

I found this resource super useful and clear: http://www.dyn4j.org/2010/01/sat/#sat-mtv
